src/engine/cachingreader/cachingreaderworker.cpp

```diff
@@ -114,27 +118,31 @@ void CachingReaderWorker::run() {
     }
 }
 
-void CachingReaderWorker::loadTrack(const TrackPointer& pTrack) {
+void CachingReaderWorker::cleanUpOldTrack() {
+    m_pAudioSource.reset(); // Close open file handles of the old track.
+
     // Discard all pending read requests
     CachingReaderChunkReadRequest request;
     while (m_pChunkReadRequestFIFO->read(&request, 1) == 1) {
         const auto update = ReaderStatusUpdate::readDiscarded(request.chunk);
         m_pReaderStatusFIFO->writeBlocking(&update, 1);
     }
+}
 
-    // Unload the track
-    m_pAudioSource.reset(); // Close open file handles
+void CachingReaderWorker::ejectTrack() {
```

Comment:
`unloadTrack()` for consistency with `loadTrack()` and because we send a *trackUnloaded* message afterwards.